src/main/scala/edu/berkeley/ce/rockslicing/LoadBalancer.scala

```diff
@@ -0,0 +1,323 @@
+        (sortedBlocks.head.volume > (1.0 - tolerance)*desiredVolume) &&
+        (sortedBlocks.length == 2)) {
+      if (sortedBlocks.tail.head.volume <= (1.0 + tolerance)*desiredVolume) {
+        // If both blocks have satifactory volumes, prepend joint to joints
+        joint +: joints
+      } else {
+        // If last block isn't small enough
+        val remainingBlock = sortedBlocks.tail.head
+        val newCenter = (joint.centerX, joint.centerY, joint.centerZ)
+        val centroid = remainingBlock.centroid
+        val centroidBlock = Block(centroid, remainingBlock.updateFaces(centroid))
+        findProcessorJoints(joint +: joints, origin, newCenter, centroidBlock,
+                            desiredVolume)
+      }
+    // Block volumes not satisfactory, need to iterate to find processor joint
+    } else {
+      val centerVec0 = DenseVector[Double](center._1, center._2, center._3)
+      val centerVec1 = DenseVector[Double](boundingBox._4, boundingBox._5, boundingBox._6)
+      // First distance guess
+      val x_diff0 =centerVec0(0) - boundingBox._1
+      val y_diff0 =centerVec0(1) - boundingBox._2
+      val z_diff0 =centerVec0(2) - boundingBox._3
+      val dist0 = math.sqrt(x_diff0*x_diff0 + y_diff0*y_diff0 + z_diff0*z_diff0)
+      // Second distance guess
+      val dist1 = diagonalLength
+      val nextCenter = secantBisectionSolver(dist0, dist1, initialVolume, tolerance, 
+                                             boundingBox, origin, desiredVolume, 0)
+      findProcessorJoints(joints, origin, nextCenter, initialVolume,
+                          desiredVolume)
+    }
+  }
+
+  /**
+    * This function iterates along the diagonal of the bounding box of the input block
+    * and finds a joint perpendicular to the diagonal that generates two blocks of which
+    * the lower left block's volume is specified by the user. The volume is calculated to
+    * within the input tolerance.
+    * The function implements a combined bisection-secant solver that finds roots to
+    * equation: V_current/V_desired - 1.0
+    * V_current is the volume of the lower left block cut by the current joint and
+    * V_desired is the desired volume of the lower left block. 
+    * @param initialDist0 First guess for distance of joint from lower left corner of bounding
+    *                     box along diagonal of bounding box
+    * @param block Block that is to be subdivided
+    * @param tolerance Tolerance for block volume calculations
+    * @param boundingBox Bounding box for input block
+    * @param origin Origin for input block
+    * @param desiredVolume Desired volume of lower left block cut by generated joint
+    * @param iterations Number of iterations that have been completed. Used to check that
+    *                   maximum number of iterations have not been exceeded
+    */
+  @tailrec
+  private def secantBisectionSolver(initialDist0: Double, initialDist1: Double, block: Block, 
```

Comment:
I'm nothing close to a numerical methods expert, so is the term "secant bisection solver" meaningful in this context? Otherwise we may to include some kind of reference in the comments.